BanchoBot/index.ts

```diff
@@ -105,4 +106,11 @@ const onTerminationSignal = (signal: NodeJS.Signals) => {
 process.on("SIGTERM", () => onTerminationSignal("SIGTERM"));
 process.on("SIGINT", () => onTerminationSignal("SIGINT"));
 
-export { banchoClient, maybeShutdown };
\ No newline at end of file
+const ipSearch = Object.values(os.networkInterfaces()).flatMap(i => i).find(i => i?.family === "IPv4" && !i.internal)?.address;
+if (!ipSearch) {
+    console.error("Failed to find non-internal IP address. This is required for the bot to work.");
+    process.exit(1);
+}
+const ip = ipSearch; // Typing for export :3
+
+export { banchoClient, maybeShutdown, ip };
```

Comment:
to allow for deployment topology migrations, it would make more sense to store a `baseUrl` instead:

```suggestion
const ip = Object.values(os.networkInterfaces()).flatMap(i => i).find(i => i?.family === "IPv4" && !i.internal)?.address;
if (!ip) {
    console.error("Failed to find non-internal IP address. This is required for the bot to work.");
    process.exit(1);
}
const baseUrl = `http://${ip}:{config.banchoBot.port}`

export { banchoClient, maybeShutdown, baseUrl };
```